### PR TITLE
Feat/axis break

### PR DESCRIPTION
**Pull request by kkxxkk2019**
[[中文版模板 / Chinese template](https://github.com/VisActor/vrender/blob/main/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Refactoring
- [ ] Update dependency
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Workflow
- [ ] Release
- [ ] Other (about what?)

### 🐞 Bugserver case id

66d673fc71a15300f7f1fb4d

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### 🚀 Summary

copilot:summary

### 🔍 Walkthrough

copilot:walkthrough

### CODE REVIEW

File: packages/vrender-components/src/axis/line.ts [hunk 1]
```diff
@@ -48,8 +56,60 @@ export class LineAxis extends AxisBase<LineAxisAttributes> {
...
+        const { range, breakSymbol, rawRange } = aBreak;
+        transformedBreaks.push({
+          startPoint: this.getTickCoord(range[0]),
+          endPoint: this.getTickCoord(range[1]),
+          range,
+          breakSymbol,
+          rawRange
+        });
+      }
+      this._breaks = transformedBreaks;
```
Comment (xile611) [inline packages/vrender-components/src/axis/line.ts:75]: 没看到清理缓存的逻辑，如果breaks 从有切换到无，会不会缓存了老的breaks

File: packages/vrender-components/src/axis/line.ts [hunk 2]
```diff
@@ -48,8 +56,60 @@ export class LineAxis extends AxisBase<LineAxisAttributes> {
...
     super._renderInner(container);
+
+    // 渲染 break symbol
+    if (this._breaks && this._breaks.length) {
+      this._breaks.forEach((b, index) => {
+        const { startPoint, endPoint, breakSymbol, rawRange } = b;
+
+        if (breakSymbol?.visible !== false) {
+          const axisBreakGroup = graphicCreator.group({
+            zIndex: 99 // 层级需要高于轴线
```
Comment (xile611) [inline packages/vrender-components/src/axis/line.ts:86]: zIndex 有通过变量统一管理吗
都是magic number，怕后续会比较乱

File: packages/vrender-components/src/axis/tick-data/continuous.ts [hunk 3]
```diff
@@ -28,20 +59,41 @@ export const continuousTicks = (scale: ContinuousScale, op: ITickDataOpt): ITick
...
+  const { tickCount, forceTickCount, tickStep, noDecimals = false, labelStyle, breakData } = op;
 
   let scaleTicks: number[];
   if (isValid(tickStep)) {
     scaleTicks = (scale as LinearScale).stepTicks(tickStep);
   } else if (isValid(forceTickCount)) {
-    scaleTicks = (scale as LinearScale).forceTicks(forceTickCount);
+    scaleTicks = getScaleTicks(op, scale, forceTickCount, (count: number, range?: [number, number]) => {
+      if (range && range.length) {
+        return (scale as LinearScale).domain(range).forceTicks(count);
```
Comment (xile611) [inline packages/vrender-components/src/axis/tick-data/continuous.ts:70]: 1. 参数2的变量名称，建议改一下，容易和scale的range混乱，建议改成 `subDomain`
2. 对breakdata 的处理要不要挪到vscale中呢？

File: packages/vrender-components/src/axis/tick-data/continuous.ts [hunk 4]
```diff
@@ -6,6 +6,37 @@ import type { ICartesianTickDataOpt, ILabelItem, ITickData, ITickDataOpt } from
...
+
+  if (breakData && breakData()) {
+    const { range: breakRanges, scope } = breakData();
+    const domain = scale.domain();
+    scaleTicks = [];
+    for (let i = 0; i < domain.length - 1; i++) {
+      const range: [number, number] = [domain[i], domain[i + 1]];
+      const ticks = getTicks(count, range); // 暂时不对个数进行分段
+      ticks.forEach(tick => {
+        if (!breakRanges.some(breakRange => tick >= breakRange[0] && tick <= breakRange[1])) {
```
Comment (xile611) [inline packages/vrender-components/src/axis/tick-data/continuous.ts:26]: // 分段后的 scale: 
scale.domain([0, 2100, 22900, 25000]).range([336, 26.879999999999995, 26.879999999999995, 0])
对于分段后的scale, 其实 2100 到 22900 这一段计算刻度，有点没意义

File: packages/vrender-components/src/axis/tick-data/continuous.ts (hunk 3 above)
Comment (kkxxkk2019) [reply packages/vrender-components/src/axis/tick-data/continuous.ts:70]: > 对breakdata 的处理要不要挪到vscale中呢？

这个先放在这里吧，目前这个ticks 的生成逻辑还不太完善，另外看了下放进 scale 里涉及到的改造点挺多的，我先加个 todo，后续完善了之后再把功能移入 scale 吧

**Review by xile611 — APPROVED** [review]

**Pull request merged**